modules/hal_nordic/nrf_802154/sl_opensource/platform/nrf_802154_temperature_zephyr.c

```diff
@@ -18,66 +18,56 @@
 #include <kernel.h>
 #include <logging/log.h>
 #include <stdbool.h>
+#include <stdint.h>
+#include <zephyr.h>
 
 #define LOG_LEVEL LOG_LEVEL_INFO
 #define LOG_MODULE_NAME nrf_802154_temperature
 LOG_MODULE_REGISTER(LOG_MODULE_NAME);
 
-/** @brief Value of the last temperature measurement in degree Celsius. */
-static int8_t temperature;
+/** @brief Structure describing the temperature measurement module. */
+static struct {
+	const struct device *dev;	/**< Temperature device pointer. */
+	int8_t value;			/**< Last temperature measurement in degree Celsius. */
 
-void nrf_802154_temperature_init(void)
-{
-	/* Intentionally empty. */
-}
+	struct k_work_delayable work;
+} temperature;
 
-void nrf_802154_temperature_deinit(void)
+static void work_handler(struct k_work *work)
 {
-	/* Intentionally empty. */
-}
+	struct sensor_value temperature_value;
 
-int8_t nrf_802154_temperature_get(void)
-{
-	return temperature;
-}
+	sensor_sample_fetch(temperature.dev);
```

Comment:
please check values returned from fetch and get.

Reply:
Values are checked and only reschedule is performed if fet of get failed.